command/agent/status_endpoint.go

```diff
@@ -42,3 +43,11 @@ func (s *HTTPServer) StatusPeersRequest(resp http.ResponseWriter, req *http.Requ
 	}
 	return peers, nil
 }
+
+func (s *HTTPServer) StatusVersionRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != "GET" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	return version.GetVersion().FullVersionNumber(true), nil
+}
```

Comment:
I went to add a commit for API docs for `/v1/status/version` and realized that this HTTP request isn't being forwarded as an RPC to the leader. So right now this is showing the version of the agent we're sending the HTTP request to (which could even be a client!), not the version of the cluster leader. Is that what we're really looking for here? If so, that information is already available in the `/v1/agent/self` API and we don't need the new HTTP endpoint:

```
$ curl -s localhost:4646/v1/agent/self | jq '.config.Version'
{
  "Revision": "c7ee53af6c88ea63165a939da5fdb17f6e784161",
  "Version": "1.2.3",
  "VersionMetadata": "",
  "VersionPrerelease": "dev"
}
```

I'd also wonder in that case if we really want to call it "server version", because that might be misleading in the case where someone is talking to a client and not a server.

If we want the actual servers, we'd want to call `/v1/agent/members` but that doesn't have the `Revision` data in the response, just the `build` tag (ex. `1.2.3-dev` in this case).